### ISSUE TITLE
use of ES5 prototype functions are not flagged as errors when parserOptions.ecmaVersion is set to 3

### ISSUE THREAD

**Issue by ispringer** [eslint/eslint#6838]
I would expect the following line to be flagged as an error by eslint if ecmaVersion is set to 3 in my config:

```
var foo = "foo".trim();
```

**Comment by gyandeeps** [comment]
Are you saying `trim` was introduced in ES5 thats why it should throw an error?
If yes, then parser's dont throw errors on prototype function for example `var foo = "foo".sdfdsf();` would parse fine.

**Comment by eslintbot** [comment]
Hi @ispringer, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
I think this is best solved by writing a plugin which will enforce this sort of thing. Unfortunately we don't have the ability to reliably determine whether something is a string or not, so it would be very difficult for us to add this sort of thing to core.

That said, if you wish to propose a new rule for us to consider, please use the Proposing a New Rule page (see link in the eslintbot comment) as guidance to flesh out a full proposal. It would probably be good to list what functions from what object types you believe should be flagged as pre-ES5, and then include examples of how each would be linted. At this point, though, we don't really have enough information to be able to help.

**Comment by ispringer** [comment]
> Are you saying trim was introduced in ES5 thats why it should throw an error?

Yeah, that's what I was hoping for. If was hoping that if I set "ecmaVersion" to 3, eslint would throw errors on usages of prototype functions on String, Object, Array, etc. that are not defined in ECMAScript 3. 

I have a project that needs to run in IE8, which is basically ES3 (though not officially), so I was hoping to set "ecmaVersion" to 3 in my config to catch usages of ES5 things like trim, which IE8 does not support.

So is "ecmaVersion" only used to check spec-defined core language features then? http://eslint.org/docs/user-guide/configuring says almost nothing about what "ecmaVersion" is used for.

@platinumazure, Yeah, that makes sense. I can see how it would be tough to determine variable types in many cases.  

Thanks!

**Comment by nzakas** [comment]
`ecmaVersion` only affects the syntax that the parser recognizes -- we have no knowledge about which methods are available on which objects. You'd need to use something like Flow or TypeScript to get that level of detail.

**Comment by ispringer** [comment]
Thanks @nzakas, I'll look into using one of those. And thanks all for clearing up my misunderstanding. 

Closing this issue...
